kolibri/core/assets/src/core-app/client.js

```diff
@@ -2,11 +2,66 @@
  * Module for REST API client
  */
 
-import rest from 'rest';
+import interceptor from 'rest/interceptor';
 import mime from 'rest/interceptor/mime';
-import csrf from 'rest/interceptor/csrf';
-import errorCode from 'rest/interceptor/errorCode';
-import cookiejs from 'js-cookie';
+import baseClient from './baseClient';
+import errorCodes from '../disconnectionErrorCodes';
+
+import heartbeat from 'kolibri.heartbeat';
+import { connected } from 'kolibri.coreVue.vuex.getters';
+import store from 'kolibri.coreVue.vuex.store';
+
+const disconnectInterceptor = interceptor({
+  request: function(request) {
+    if (!connected(store.state)) {
+      // If the vuex state records that we are not currently connected then cancel all
+      // outgoing requests.
+      if (request.cancel) {
+        // If the request has a cancel method, call it.
+        request.cancel();
+      }
+      // Set the request as canceled
+      request.canceled = true;
+    }
+    return request;
+  },
+  response: function(response) {
+    if (response.request && response.request.canceled) {
+      // If a request has been canceled, reject the response and throw an error
+      return Promise.reject(response);
+    }
+    return response;
+  },
+});
+
+const loginTimeoutDetection = interceptor({
```

Comment:
Could we rename this to `loggedInTimeoutDetection` or `sessionTimeoutDetection`?